src/plans/D3MMIP21Plan.sol

```diff
@@ -0,0 +1,156 @@
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.14;
+
+import "./ID3MPlan.sol";
+
+interface TokenLike {
+    function totalSupply() external view returns (uint256);
+    function transfer(address, uint256) external returns (bool);
+    function transferFrom(address, address, uint256) external returns (bool);
+    function balanceOf(address) external view returns (uint256);
+}
+
+/// Combines the MIP21 Input and Output conduits
+contract D3MMIP21Plan is ID3MPlan {
+
+    mapping (address => uint256) public wards;
+    mapping (address => uint256) public can;
+    mapping (address => uint256) public bud;
+    address                      public to;
+    uint256                      public cap;
+
+    TokenLike public immutable dai;
+    TokenLike public immutable gov;
+    address public immutable escrow;
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, uint256 data);
+    event Hope(address indexed usr);
+    event Nope(address indexed usr);
+    event Kiss(address indexed who);
+    event Diss(address indexed who);
+    event Pick(address indexed who);
+    event Push(address indexed to, uint256 wad);
+
+    constructor(address dai_, address gov_, address escrow_) {
+        dai = TokenLike(dai_);
+        gov = TokenLike(gov_);
+        escrow = escrow_;
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth {
+        require(wards[msg.sender] == 1, "D3MMIP21Plan/not-authorized");
+        _;
+    }
+
+    modifier operator {
+        require(can[msg.sender] == 1, "D3MMIP21Plan/not-operator");
+        _;
+    }
+
+    // --- Math ---
+    function _min(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = x <= y ? x : y;
+    }
+
+    // --- Admin ---
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    function hope(address usr) external auth {
+        can[usr] = 1;
+        emit Hope(usr);
+    }
+    function nope(address usr) external auth {
+        can[usr] = 0;
+        emit Nope(usr);
+    }
+
+    function file(bytes32 what, uint256 data) external auth {
```

Comment:
Might actually want this to be the `operator` modifier.  This would allow governance to set an `ilk.line` or `vat.Line` to limit exposure, and the `operator` to set `cap` to the maximum debt they want to pull (or us to inject) at any given time.